content/blog/2024-07-07-aws-to-contribute-efficiency-improvements-for-ValKey-8.md

```diff
@@ -0,0 +1,83 @@
++++
+# `title` is how your post will be listed and what will appear at the top of the post
+title= "Unlock 1 Million RPS: Experience Triple the Speed with Valkey"
+# `date` is when your post will be published.
+# For the most part, you can leave this as the day you _started_ the post.
+# The maintainers will update this value before publishing
+# The time is generally irrelvant in how Valkey published, so '01:01:01' is a good placeholder
+date= 2024-07-08 01:01:01
+# 'description' is what is shown as a snippet/summary in various contexts.
+# You can make this the first few lines of the post or (better) a hook for readers.
+# Aim for 2 short sentences.
+description= "Better efficiency reduces cost, improves latency and makes our environment greener"
+# 'authors' are the folks who wrote or contributed to the post.
+# Each author corresponds to a biography file (more info later in this document)
+authors= [ "dantouitou", "uriyagelnik"]
++++
+Valkey 8.0, set for release in September 2024, brings major performance enhancements through improved multi-threading.
+This update aims to significantly boost throughput and reduce latency across various hardware configurations.
+Read on to learn more about the upcoming features in Valkey 8.0, including the innovative I/O threading implementation and its impact on performance and efficiency.
+
+### Our Commitment to performance and efficiency
+At AWS, we have hundreds of thousands of customers using Amazon ElastiCache and Amazon MemoryDB.
+Feedback we continuously hear from end users is that they need better absolute performance and want to squeeze more performance from their clusters.
+
+For absolute performance per node, users sometimes face challenges with hot keys or other limitations that prevent them from scaling out.
+They also need better latencies at high percentiles.
+For efficiency, users are constantly trying to optimize their costs and get more out of their hardware.
+
+Besides performance, efficiency is also seen as a way to improve resilience.
+For example, with fork-based snapshotting and full-sync replication, the longer it takes, the more memory is consumed by copied-on-write pages and the replica output buffer, often resulting in high swapping and erratic behavior.
+By reducing execution time, one can achieve a more predictable memory usage and greater stability.
```

Comment:
I still find these two paragraphs really awkward and a lot of what they are saying is already explained elsewhere. The second one is especially a tangent for fork-based snapshot. It doesn't feel like it fits into the narrative well. I know Itay wanted to include reliability in the narrative, I just don't think we are covering it well enough to explain it. Maybe I'll wait for Itay to comment about it.

Reply:
In addition to hotkeys, Valkey users with multi-key operations (mget and co) have to ensure that all keys in such operations are located in the same slot. In many cases, the result is that they are using a small number of slots and consequently are restricted on how much their cluster can scale. These users are often looking for larger , more powerful shards

Reply:
I guess after reflecting on it some more, I'm inclined to say we should really lean in on the single key/slot use case and say we are accelerating that. We already have a section dedicated to the benefit of bigger shards, so maybe we can move the first paragraph + Dan's added use case there.

Reply:
I have removed the sentences about performance and efficiency and added Dan's wording to the "Benefits of High Capacity Shards" section.